src/creature.cpp

```diff
@@ -1192,14 +1192,40 @@ void Creature::deal_damage_handle_type( const damage_unit &du, bodypart_id bp, i
 
         case DT_HEAT:
             // heat damage sets us on fire sometimes
-            if( rng( 0, 100 ) < adjusted_damage ) {
+            // Volatile enemies always ignite from fire damage, may take increased damage
+            if( rng( 0, 100 ) < adjusted_damage || has_flag( MF_VOLATILE ) ) {
                 add_effect( effect_onfire, rng( 1_turns, 3_turns ), bp.id() );
+                if( has_flag( MF_VOLATILE ) && one_in( 4 ) ) {
+                    if( g->u.sees( *this ) ) {
+                        add_msg( m_warning, _( "The %s goes up in flames!" ), get_name() );
+                    }
+                    adjusted_damage *= 10;
+                }
```

Comment:
this code is duplicated accross 3 places, extract into lambda to something like:
```suggestion
auto set_fire = [&, this]( int chances ) {
    if( has_flag( MF_VOLATILE ) && one_in( chances ) ) {
        add_effect( effect_onfire, rng( 1_turns, 3_turns ), bp.id() );
        if( get_avatar().sees( *this ) ) {
            add_msg( m_warning, _( "The %s goes up in flames!" ), get_name() );
        }
        adjusted_damage *= 10;
    }
};                
```